composer/trainer/trainer.py

```diff
@@ -285,27 +279,19 @@ def __init__(
                     To fix, please do not iterate over the dataloader before passing it into
                     the trainer."""))
 
-        if eval_subset_num_batches is not None:
```

Comment:
Curious why this was removed?

Reply:
It's a pretty verbose check for a narrow issue. I think we can clarify in the docs instead of throw a warning.